lib/fenrir_view/presenter.rb

```diff
@@ -24,6 +25,29 @@ def partial
 
     private
 
+    def validate_properties
+      class_properties = self.class._properties
+      instance_properties = properties
+      return unless class_properties.any?
+
+      if !instance_properties.is_a?(Hash) && instance_properties != false
+        raise "An instance of #{slug} has properties as #{instance_properties.class.name}. Should be Hash or false."
+      elsif class_properties.keys != instance_properties.keys
+        unknown_keys = instance_properties.keys.reject { |key| class_properties.include?(key) }
+        raise "#{self.class.name} has unkown keys: #{unknown_keys.join(', ')}"
+      end
+
+      instance_properties.each do |property, value|
+        property_validations = class_properties[property]
+
+        if property_validations
+          raise "An instance of #{slug} is missing the required property: #{property}" if value.blank? && property_validations[:required].presence
+          raise "An instance of #{slug} has the wrong type: '#{value.class}' for property: '#{property}' (value: '#{value}'). Should be one of: #{property_validations[:one_of_type]}" unless property_validations[:one_of_type]&.include?(value.class)
```

Comment:
This unless nilcheck business is probably not anything we want

Reply:
What if the value is missing, but it's not required? 
Should it be required if it needs a type of be one of a list?

Reply:
This is super hard to read. Might be worthwhile abstracting each raise case to a method?

Reply:
Yeah, I did try that initially, but now that I've had a better look at it it's probably time to try it again 👍